### PR TITLE
feat: [CDS-54179]: ECS NG Rolling, Blue Green Outcome

### PR DESCRIPTION
**Pull request by alluvamsi-harness**
## Describe your changes
Added serviceName at First level in ECS Step Output
<+pipeline.stages.ecs.spec.execution.steps.Rolling.output.serviceName>
image
Backward compatibility
The serviceName will be passed as null and the Outcome will not be shown in Output
image

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44429)

### CODE REVIEW

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsRollingDeployStep.java [hunk 1]
```diff
@@ -106,12 +108,22 @@ public StepResponse finalizeExecutionWithSecurityContext(Ambiance ambiance, Step
     if (ecsRollingDeployResponse.getCommandExecutionStatus() != CommandExecutionStatus.SUCCESS) {
       return EcsStepCommonHelper.getFailureResponseBuilder(ecsRollingDeployResponse, stepResponseBuilder).build();
     }
+    EcsRollingDeployOutcome ecsRollingDeployOutcome =
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsRollingDeployStep.java:111]: What if no tasks are present or created?  Did we test that case?

**Review by sainath-b — CHANGES_REQUESTED** [review]
Please discuss and address comments

**Comment by alluvamsi-harness** [comment]
trigger smartchecks

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — COMMENTED** [review]
> Reviewed 2 of 3 files at r1, 1 of 2 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44429)* status: 7 file(s) to review. 1 of 2 approvals obtained.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsRollingDeployStep.java (hunk 1 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsRollingDeployStep.java:112]: remove it

**Comment by alluvamsi-harness** [comment]
trigger messagecheck

**Review by rohitk-h — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 2 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44429)* status: 1. unresolved discussions.

**Review by utkarshchoubey-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44429)* status: 1. unresolved discussions.

**Review by yogesh-chauhan — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44429)* status: :shipit: complete! ✓

**Pull request merged**